huaweicloud/services/dms/resource_huaweicloud_dms_rocketmq_topic.go

```diff
@@ -0,0 +1,352 @@
+				Computed:     true,
+				Description:  `Specifies the permissions of the topic.`,
+				ValidateFunc: validation.StringInSlice([]string{"all", "sub", "pub"}, false),
+			},
+			"read_queue_num": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Computed:    true,
+				Description: `Specifies the total number of read queues.`,
+			},
+			"write_queue_num": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Computed:    true,
+				Description: `Specifies the total number of write queues.`,
+			},
+		},
+	}
+}
+
+func DmsRocketMQTopicBrokerRefSchema() *schema.Resource {
+	sc := schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"broker_name": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: `Indicates the name of the broker.`,
+			},
+			"read_queue_num": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Computed:    true,
+				Description: `Indicates the read queues number of the broker.`,
+			},
+			"write_queue_num": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Computed:    true,
+				Description: `Indicates the read queues number of the broker.`,
+			},
+		},
+	}
+	return &sc
+}
+
+func resourceDmsRocketMQTopicCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	region := config.GetRegion(d)
+
+	// createRocketmqTopic: create DMS rocketmq topic
+	var (
+		createRocketmqTopicHttpUrl = "v2/{project_id}/instances/{instance_id}/topics"
+		createRocketmqTopicProduct = "dms"
+	)
+	createRocketmqTopicClient, err := config.NewServiceClient(createRocketmqTopicProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating DmsRocketMQTopic Client: %s", err)
+	}
+
+	instanceID := d.Get("instance_id").(string)
+	createRocketmqTopicPath := createRocketmqTopicClient.Endpoint + createRocketmqTopicHttpUrl
+	createRocketmqTopicPath = strings.ReplaceAll(createRocketmqTopicPath, "{project_id}", createRocketmqTopicClient.ProjectID)
+	createRocketmqTopicPath = strings.ReplaceAll(createRocketmqTopicPath, "{instance_id}", instanceID)
+
+	createRocketmqTopicOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+	}
+	createRocketmqTopicOpt.JSONBody = utils.RemoveNil(buildCreateRocketmqTopicBodyParams(d, config))
+	createRocketmqTopicResp, err := createRocketmqTopicClient.Request("POST", createRocketmqTopicPath, &createRocketmqTopicOpt)
+	if err != nil {
+		return diag.Errorf("error creating DmsRocketMQTopic: %s", err)
+	}
+
+	createRocketmqTopicRespBody, err := utils.FlattenResponse(createRocketmqTopicResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	id, err := jmespath.Search("id", createRocketmqTopicRespBody)
+	if err != nil {
+		return diag.Errorf("error creating DmsRocketMQTopic: ID is not found in API response")
+	}
+	d.SetId(instanceID + "/" + id.(string))
+
+	return resourceDmsRocketMQTopicRead(ctx, d, meta)
```

Comment:
you can return resourceDmsRocketMQTopicUpdate here,
so that read_queue_num and write_queue_num work in create

Reply:
the read method can get the data also, and the value is equal to queue_num

Reply:
done